src/hotspot/share/jfr/recorder/storage/jfrMemorySpaceRetrieval.hpp

```diff
@@ -85,7 +85,6 @@ class JfrMspaceRemoveRetrieval : AllStatic {
       Node* const node = iterator.next();
       if (node == nullptr) return nullptr;
       assert(node->free_size() >= size, "invariant");
-      assert(!node->retired(), "invariant");
       assert(node->identity() == nullptr, "invariant");
```

Comment:
This is not the right way to do it.

Reply:
Good catch, I removed it during debugging.